### PR TITLE
Fixed border radius not applying properly and also added missing localizations for action buttons

### PR DESCRIPTION
**Pull request by gidrokolbaska**

### CODE REVIEW

File: lib/src/widgets/calendar_date_picker2_with_action_buttons.dart [hunk 1]
```diff
@@ -23,6 +25,12 @@ class CalendarDatePicker2WithActionButtons extends StatefulWidget {
   /// The calendar configurations including action buttons
   final CalendarDatePicker2WithActionButtonsConfig config;
 
+  /// The text that is displayed on the cancel button.
```
Comment (theideasaler) [inline lib/src/widgets/calendar_date_picker2_with_action_buttons.dart:28]: Users can use `customCancelButton` & `customOkButton` for the customisation. 

Please revert these changes.

File: lib/src/widgets/calendar_date_picker2_with_action_buttons.dart [hunk 2]
```diff
@@ -117,7 +129,9 @@ class _CalendarDatePicker2WithActionButtonsState
             const EdgeInsets.symmetric(vertical: 10, horizontal: 15),
         child: widget.config.cancelButton ??
             Text(
-              'CANCEL',
+              Theme.of(context).useMaterial3
```
Comment (theideasaler) [inline lib/src/widgets/calendar_date_picker2_with_action_buttons.dart:132]: Please keep `localizations.cancelButtonLabel.toUpperCase()` only

File: lib/src/widgets/calendar_date_picker2_with_action_buttons.dart [hunk 3]
```diff
@@ -147,7 +162,7 @@ class _CalendarDatePicker2WithActionButtonsState
             const EdgeInsets.symmetric(vertical: 10, horizontal: 15),
         child: widget.config.okButton ??
             Text(
-              'OK',
+              localizations.okButtonLabel,
```
Comment (theideasaler) [inline lib/src/widgets/calendar_date_picker2_with_action_buttons.dart:165]: Please change to `localizations. okButtonLabel.toUpperCase()`

File: lib/src/widgets/calendar_date_picker2_with_action_buttons.dart [hunk 4]
```diff
@@ -75,8 +83,11 @@ class _CalendarDatePicker2WithActionButtonsState
 
   @override
   Widget build(BuildContext context) {
+    final MaterialLocalizations localizations =
+        MaterialLocalizations.of(context);
     return Column(
-      mainAxisSize: MainAxisSize.min,
+      mainAxisSize: MainAxisSize.max,
```
Comment (theideasaler) [inline lib/src/widgets/calendar_date_picker2_with_action_buttons.dart:89]: Please revert changes for `Column` widget

File: lib/src/utils/dialog.dart [hunk 5]
```diff
@@ -20,23 +20,22 @@ Future<List<DateTime?>?> showCalendarDatePicker2Dialog({
 }) {
   var dialog = Dialog(
```
Comment (theideasaler) [inline lib/src/utils/dialog.dart:21]: @gidrokolbaska Could you please attach a screenshot demonstrating what is the `borderRadius` issue?

File: lib/src/widgets/calendar_date_picker2_with_action_buttons.dart (hunk 1 above)
Comment (gidrokolbaska) [reply lib/src/widgets/calendar_date_picker2_with_action_buttons.dart:28]: Yeah, my bad. Forgot to delete those fields. They are not used anywhere anyway. Will answer all the remaining comments later today

**Comment by gidrokolbaska** [comment]
Made another commit to reflect the changes based on your feedback

File: lib/src/utils/dialog.dart (hunk 5 above)
Comment (gidrokolbaska) [reply lib/src/utils/dialog.dart:21]: Sure!
So this is how borderRadius works by default:
image
And here is the code:
```dart
await showCalendarDatePicker2Dialog(
              config: CalendarDatePicker2WithActionButtonsConfig(
                controlsHeight: 170,
                lastDate: DateTime.now(),
                firstDate: DateTime(DateTime.now().year - 2),
                calendarType: CalendarDatePicker2Type.single,
              ),
              context: context,
              borderRadius: BorderRadius.circular(0),// low border radius is not taken into consideration
              dialogSize: Size(86.w, 66.h),
            );
```
It only begins to works if you set the borderRadius to some high value, e.g. 90:
image


And with my fix, you are able to set the border radius that you actually desire:
image

File: lib/src/widgets/calendar_date_picker2_with_action_buttons.dart [hunk 6]
```diff
@@ -75,8 +75,11 @@ class _CalendarDatePicker2WithActionButtonsState
 
   @override
   Widget build(BuildContext context) {
+    final MaterialLocalizations localizations =
+        MaterialLocalizations.of(context);
     return Column(
       mainAxisSize: MainAxisSize.min,
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
```
Comment (theideasaler) [inline lib/src/widgets/calendar_date_picker2_with_action_buttons.dart:82]: Please also revert this - this might be redundant

File: lib/src/utils/dialog.dart (hunk 5 above)
Comment (theideasaler) [reply lib/src/utils/dialog.dart:21]: Hi @gidrokolbaska thanks for resolving my comments, the localisations part looks great! 

However, in regard to the BorderRadius issue, I couldn't reproduce this issue on my end. If you think this is a real bug, could you please raise a bug and attach the minimum reproducible repo/code for this issue?

You could revert the border radius changes in this PR so I can merge the localization changes first, and raise another PR later for the BorderRadius if that can be confirmed as a real bug.

Comment (gidrokolbaska) [reply lib/src/utils/dialog.dart:21]: Aha, I might know why you couldn't reproduce it. Set useMaterial3 to **true** in your ThemeData class, then you will be able to reproduce the issue

Comment (theideasaler) [reply lib/src/utils/dialog.dart:21]: thanks, I can see the issue now

**Review by theideasaler — APPROVED** [review]
